lib/greed/roll_score.rb

```diff
@@ -0,0 +1,62 @@
+module Greed
+  class RollScore
+    attr_reader :value
+
+    def initialize(dice)
+      raise ArgumentError, "Valid roll object not provided" unless dice
+      @dice = dice
+      @value = calculate_score
+    end
+
+    def non_scoring_dice
```

Comment:
I can see the logic in moving
`if dice_array.size == 0 then dice_array = Array.new(5) { Die.new } end`
outside of the RollScore module.

I could set non_scoring_dice as an instance variable, and change this to a private method that sets it (which would be called in the initialize). Does that sound like a more appropriate solution?